### ISSUE TITLE
Ambiguous exports in CE3: Sync, etc.

### ISSUE THREAD

**Issue by arosien** [typelevel/cats-effect#1290]
I have

```scala
import cats.effect._
import cats.effect.kernel._
```

and get

```
reference to Sync is ambiguous;
it is imported twice in the same scope by
import cats.effect.kernel._
and import cats.effect._
```

I'm doing import cats.effect.kernel._ to get `Ref` into scope, and this is triggered from `def foo[F[_]: Sync] = ...`.

@SystemFw says:

> we reexport it, but that might be problematic
> we should either reexport Ref and Deferred too, and discourage the use of kernel, or remove reexports completely, I guess

**Comment by vasilmkd** [comment]
I'm guessing the reexports are there to ease transition from CE2? If that's the case, we cannot really get rid of them, so we should probably reexport everything then.

**Comment by djspiewak** [comment]
> I'm guessing the reexports are there to ease transition from CE2? If that's the case, we cannot really get rid of them, so we should probably reexport everything then.

Both the transition from CE2 and also to reduce the cognitive load on users who are just depending on **core**. The latter in particular is pretty strong here; I think we should reexport `Ref` and `Deferred` within the `cats.effect`. package object, but the rest of the `std` stuff should remain within its own package.

**Comment by vasilmkd** [comment]
Resolved with #1339.
